enhancements/vdpa/virtio-vdpa-ovn-hw-offload.md

```diff
@@ -0,0 +1,235 @@
+---
+title: virtio-vdpa-ovn-hw-offload
+authors:
+  - Leonardo Milleri
+  - Adrian Moreno Zapata
+  - Ariel Adam
+reviewers:
+  - "@mandre"
+  - "@tjungblu"
+  - "@zshi-redhat"
+  - "@SchSeba"
+  - "@bn222"
+approvers:
+  - "@dcbw"
+  - "@knobunc"
+api-approvers:
+  - "@dcbw"
+  - "@knobunc"
+creation-date: 2022-07-13
+last-updated: 2022-07-29
+tracking-link: 
+  - https://issues.redhat.com/browse/NP-19
+---
+
+
+# Virtio/vDPA with OVN HW offload
+
+The purpose of this enhancement is to provide a proposal for integrating Virtio/vDPA in OCP for container workloads.
+
+
+## Summary
+See [VDPA support in OCP (Overview)](https://github.com/openshift/enhancements/blob/master/enhancements/vdpa/vdpa-support-overview.md).
+
+Main aspects of this enhancement:
+- primary interface in the pod is configured as a Virtio/vDPA device
+- NIC configured in switchdev mode
+- OVS HW offloading enabled
+- Supported HW: NVIDIA CTX-6 DX
+
+
+## Motivation
+
+The main motivation is to integrate a Virtio/vDPA device in Openshift.
+
+
+### User Stories
+
+Details are provided in the overview enhancement.
+
+### Goals
+
+- As a user I want to configure an Openshift cluster with Virtio/VDPA support.
+- As a user I want to determine how many VFs to create for a given NIC and how each VF should be consumed by workloads (e.g., netdev device in the container network namespace)
+- As a user I want the VDPA solution to be integrated with OVN-kubernetes and with HW offloading enabled
+
+### Non-Goals
+
+- The Vhost-vdpa driver won't be supported in CNFs (VMs only). The virtio-vdpa driver will be supported in CNFs, i.e. netdev device to be consumed.
+- Scalable IOV and Sub-functions are out of scope, only switchdev SR-IOV will be supported (with OVS HW offloading to the NIC)
+
+## Proposal
+
+The following diagram depicts all the involved components and their interactions:
```

Comment:
step 4 in the diagram should be done by the sriov-operator not the sriov device plugin like we have today with the vhost-net and tun device 

https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/pkg/daemon/daemon.go#L209

Reply:
Updating diagram: step 4 will be included in step 1, adding port representors and OVS